lib/model/services/ble/machine_service.dart

```diff
@@ -716,6 +728,22 @@ class EspressoMachineService extends ChangeNotifier {
               // }
             }
           }
+
+          if (scaleService.state[0] == ScaleState.connected) {
+            var frame = profileService.currentProfile?.shotFrames[shot.frameNumber];
+            //var stepWeightLimit = state.shotFrame?.maxWeight ?? 0.0;
+            var stepWeightLimit = frame?.maxWeight ?? 0.0;
```

Comment:
I found that `state.shotFrame` isn't updated over the course of the shot–maybe it should instead be a getter that uses this logic?